packages/validators/src/schemas/v2023_09_28/experience.ts

```diff
@@ -0,0 +1,186 @@
+import { z } from 'zod';
+import { SchemaVersions } from '../schemaVersions';
+
+const propertyKeySchema = z.string().regex(/^[a-zA-Z0-9-_.]{1,64}$/);
+
+const DataSourceSchema = z.record(
+  propertyKeySchema,
+  z.object({
+    sys: z.object({
+      type: z.literal('Link'),
+      id: z.string(),
+      linkType: z.literal('Entry').or(z.literal('Asset')),
+    }),
+  })
+);
+
+const PrimitiveValueSchema = z.union([
+  z.string(),
+  z.boolean(),
+  z.number(),
+  z.record(z.any(), z.any()),
+  z.undefined(),
+]);
+
+export const ComponentDefinitionPropertyTypeSchema = z.enum([
+  'Text',
+  'RichText',
+  'Number',
+  'Date',
+  'Boolean',
+  'Location',
+  'Media',
+  'Object',
+]);
+
+const ValuesByBreakpointSchema = z.record(z.lazy(() => PrimitiveValueSchema));
+
+const ComponentPropertyValueSchema = z.union([
+  z.object({
+    type: z.literal('DesignValue'),
+    valuesByBreakpoint: ValuesByBreakpointSchema,
+  }),
+  z.object({
+    type: z.literal('BoundValue'),
+    path: z.string(),
+  }),
+  z.object({
+    type: z.literal('UnboundValue'),
+    key: z.string(),
+  }),
+  z.object({
+    type: z.literal('ComponentValue'),
+    key: z.string(),
+  }),
+]);
+
+export type ComponentPropertyValue = z.infer<typeof ComponentPropertyValueSchema>;
+
+const BreakpointSchema = z.object({
+  id: z.string(),
+  query: z.string(),
+  previewSize: z.string(),
+  displayName: z.string(),
+});
+
+const UnboundValuesSchema = z.record(
+  propertyKeySchema,
+  z.object({
+    value: PrimitiveValueSchema,
+  })
+);
+
+const baseComponentTreeNodeSchema = z.object({
+  definitionId: z.string(),
+  variables: z.record(propertyKeySchema, ComponentPropertyValueSchema),
+});
+
+export type ComponentTreeNode = z.infer<typeof baseComponentTreeNodeSchema> & {
+  children: ComponentTreeNode[];
+};
+
+const ComponentTreeNodeSchema: z.ZodType<ComponentTreeNode> = baseComponentTreeNodeSchema.extend({
+  children: z.lazy(() => ComponentTreeNodeSchema.array()),
+});
```

Comment:
[Aligning the casing for schema variables with a capital letter]
I guess the "base" schema is a workaround as recursiveness doesn't play nicely with zod and type inference? If that's true, I added a comment already to make this clear.
```suggestion
// Use helper schema to define a recursive schema with its type correctly below
const BaseComponentTreeNodeSchema = z.object({
  definitionId: z.string(),
  variables: z.record(propertyKeySchema, ComponentPropertyValueSchema),
});

export type ComponentTreeNode = z.infer<typeof BaseComponentTreeNodeSchema> & {
  children: ComponentTreeNode[];
};

const ComponentTreeNodeSchema: z.ZodType<ComponentTreeNode> = BaseComponentTreeNodeSchema.extend({
  children: z.lazy(() => ComponentTreeNodeSchema.array()),
});
```